### ISSUE TITLE
Scala 2.12 compatible release

### ISSUE THREAD

**Issue by pjan** [kamon-io/Kamon#410]
As per the title? Anything on the horizon? Anything blocking this?

**Comment by henders** [comment]
Looks like its blocked on 2.12 releases for:
```
com.twitter#util-core
io.spray#spray-can
io.spray#spray-client
io.spray#spray-routing
net.virtual-void#json-lenses
org.fluentd#fluent-logger-scala
```

**Comment by lustefaniak** [comment]
Spray 2.12.0 build is not planned: https://github.com/spray/spray/issues/1132

**Comment by ivantopo** [comment]
oh boy, we need to separate our projects! We internally talked about it a lot, having each sub-project on their own repo allows them to evolve independently, we could make sure that Spray gets support for the cool features included in upcoming Kamon versions without having to have a Spray for Scala 2.12, and kamon-core could evolve without problems. The other side of that is that we will have to have some sort of compatibility rules, so that users can know what versions of sub projects work with what versions of Kamon.. it has been postponed (procrastinated, if you will) but its time to do this.

Thanks for bringing this topic up, we will do something about it!

**Comment by pjan** [comment]
Maybe a great time to move toa build.sbt based project definition as well, since the project-based one will get deprecated.

Anything that the community could do with helping in this effort/these efforts?

**Comment by ivantopo** [comment]
one thing that would be awesome is to be able to move things to another repositories but keeping the history of the files that will remain there.. that's something I don't know how to achieve and will need to research a bit on how to do it, do you have any ideas?

**Comment by henders** [comment]
@ivantopo , You can use `git filter-branch`. Examples here: http://blogs.atlassian.com/2014/04/tear-apart-repository-git-way/

**Comment by henders** [comment]
Sorry, better official example: https://help.github.com/articles/splitting-a-subfolder-out-into-a-new-repository/

**Comment by dpsoft** [comment]
Tasks that we need complete in order to achieve a Scala 2.12 compatible release:
- [ ] Separate our projects in their own repositories. 
     * Separated in repos:
          * Play
          * Backends
          * Spray
          * Akka
          * Core, Annotation and Scala in Master repo?
- [ ] Move to a ```build.sbt``` based project?
- [ ] Remove Scala 2.10 support?
- [ ] ????

thougths?

**Comment by pjan** [comment]
I believe only the first tick is required for 2.12 compatible releases

what would help to get this started, is if these repo's exist for starters

**Comment by leachbj** [comment]
Is it possible to set the scala version on the sub-projects so that kamon-spray can be skipped?  That would be a lot easier than breaking up the repo.

**Comment by dpsoft** [comment]
@leachbj @pjan @lustefaniak @henders @agarbutt @ivantopo I've moved all the repositories and the supported releases and dependencies are shown below.

| kamon  | status | jdk  | scala            | akka   |
|:------:|:------:|:----:|------------------|:------:|
|  0.7.0 | stable | 1.8+ | 2.11, 2.12       | 2.4.x  |
|  0.6.4 | stable | 1.7+ | 2.10, 2.11       | 2.3.x  |

The series 0.7.x will be live in master repository and the series 0.6.x in kamon-0.6.x repository.
We just need merge the kamon clean-master rempository to the current master repo.

thoughts?

**Comment by dpsoft** [comment]
closed by 0231d5fef3b59fe53041d950bece0c001522749c

**Comment by JeroenMinnaert** [comment]
@dpsoft I cannot seem to find a version of kamon-play2.5 for Scala 2.12. Has this ever been built?

**Comment by dpsoft** [comment]
Play 2.5 is not compatible with Scala 2.12.

Cheers,
Diego

**Comment by JeroenMinnaert** [comment]
Thanks @dpsoft, any time frame on Play 2.6 support (released today)?

**Comment by dpsoft** [comment]
Currently we are a PR with this :
https://github.com/kamon-io/kamon-play/pull/10 I think that we Will merge
un weekend 😀
